hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/ContainerEndpoint.java

```diff
@@ -141,10 +140,18 @@ public Response getContainers(
       // Send back an empty response
       return Response.status(Response.Status.NOT_ACCEPTABLE).build();
     }
+    if (prevKey > 0) {
+      // Increase the limit by 1 to fetch one additional container
+      // since we are excluding the container with the same ID as prevKey
+      limit = limit + 1;
+    }
+
     long containersCount;
-    Collection<ContainerMetadata> containerMetaDataList =
+    List<ContainerMetadata> containerMetaDataList =
         containerManager.getContainers(ContainerID.valueOf(prevKey), limit)
             .stream()
+            // Exclude the container with the same ID as prevKey
+            .filter(container -> container.getContainerID() != prevKey)
```

Comment:
By increasing prevKey by one in the before step we don't need to do filter here as the results won't contain prevKey container info. And by doing this we get actual number of container information and not one extra container detail.

Reply:
Done thanks for the suggestion !!